target_snowflake/__init__.py

```diff
@@ -170,6 +171,31 @@ def persist_lines(config, lines, table_cache=None, file_format_type: FileFormatT
             else:
                 records_to_load[stream][primary_key_string] = o['record']
 
+            if config.get('archive_load_files.enabled'):
```

Comment:
we also need to check if we're using the `S3UploadClient` and not the `SnowflakeUploadClient`. Archiving load files is acceptable only if we use the `S3UploadClient`.

The `SnowflakeUploadClient` is using snowflake managed table stages and we can't archive anything onto it.

There is a check at around https://github.com/transferwise/pipelinewise-target-snowflake/blob/ad0d71baa9dee4c47b05869b7ca6c333593f8da3/target_snowflake/db_sync.py#L273